packages/wrangler/src/init.ts

```diff
@@ -620,3 +706,78 @@ async function findPath(
 		});
 	}
 }
+
+async function createWorkerConfig(
+	accountId: string,
+	fromDashScriptName: string,
+	environment = "production"
+) {
+	// use Promise.all to fetch in parallel
+	const [bindings, routes, serviceEnvMetadata, cronTriggers] =
+		await Promise.all([
+			fetchResult<BindingsRes>(
+				`/accounts/${accountId}/workers/services/${fromDashScriptName}/environments/${environment}/bindings`
+			),
+			fetchResult<RoutesRes>(
+				`/accounts/${accountId}/workers/services/${fromDashScriptName}/environments/${environment}/routes`
+			),
+			fetchResult<ServiceMetaDataRes["default_environment"]>(
+				`/accounts/${accountId}/workers/services/${fromDashScriptName}/environments/${environment}`
+			),
+			fetchResult<CronTriggersRes>(
+				`/accounts/${accountId}/workers/scripts/${fromDashScriptName}/schedules`
+			),
+		]);
```

Comment:
What does it look like when any of these services fails/errors?

Reply:
I was thinking about that 🤔 , would `.allSettled` make more sense or should maybe early exit with error messaging?